site/src/pages/WorkspacePage/WorkspaceActions/DownloadLogsDialog.tsx

```diff
@@ -0,0 +1,166 @@
+import { useTheme, type Interpolation, type Theme } from "@emotion/react";
+import Skeleton from "@mui/material/Skeleton";
+import { saveAs } from "file-saver";
+import JSZip from "jszip";
+import { useMemo, useState, type FC } from "react";
+import { useQueries, useQuery } from "react-query";
+import { agentLogs, buildLogs } from "api/queries/workspaces";
+import type { Workspace, WorkspaceAgent } from "api/typesGenerated";
+import {
+  ConfirmDialog,
+  type ConfirmDialogProps,
+} from "components/Dialogs/ConfirmDialog/ConfirmDialog";
+import { displayError } from "components/GlobalSnackbar/utils";
+import { Stack } from "components/Stack/Stack";
+
+type DownloadLogsDialogProps = Pick<
+  ConfirmDialogProps,
+  "onConfirm" | "onClose" | "open"
```

Comment:
Should we omit `onConfirm` since we always override it?

Reply:
Maybe... will take a look.